### PR TITLE
OnCameraIdle hook into quickzoom gesture

### LINKED ISSUE #9320: Quick zoom end doesn't trigger OnCameraIdle
With the upcoming 5.1.0 release, we are introducing a new system around camera notifications  (see #8644). Currently the `quick zoom` gesture, which is performed by double-tap dragging the map, doesn't trigger the `OnCameraIdle` callback when the gesture ends.

### PR DESCRIPTION
**Pull request by tobrun**
closes #9320 , refs #9335. This PR improves the OnCameraIdle integration. Need to do some tests around [this](https://github.com/mapbox/mapbox-gl-native/pull/9339/files#diff-dc8b8b458b4be05d4cfad1bbafea82aeR395) addition. There are some cases where it's hard to trigger a rotation due to checking that flag. Testing of this feature can be done in CameraPositionActivity.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java [hunk 1]
```diff
@@ -553,7 +566,7 @@ public boolean onRotate(RotateGestureDetector detector) {
       // If rotate is large enough ignore a tap
       // Also is zoom already started, don't rotate
       totalAngle += detector.getRotationDegreesDelta();
-      if (!zoomStarted && ((totalAngle > 20.0f) || (totalAngle < -20.0f))) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java]: @Guardiola31337 the removal of zoomstarted indicates that both scale as rotation can occur at the same time. I tried reimplementing to make it work but introduced too many changes. I'm punting on this for now, will write a follow up ticket.

**Comment by tobrun** [comment]
@Guardiola31337 I revisited this PR, would you be able to eyes and test this?

File: platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/CameraPositionActivity.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 
   private MapView mapView;
   private MapboxMap mapboxMap;
```
Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/CameraPositionActivity.java]: > Private field 'mapboxMap' is assigned but never accessed

Comment (tobrun) [inline platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/camera/CameraPositionActivity.java]: That is a requirement for our autogenerated tests, see https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/utils/OnMapReadyIdlingResource.java#L43 where we use reflection on the mapboxMap field. In this specific case we could easily make it accessed by mapboxMap invocations below.

**Pull request merged**